File: java/AdvancedContactsApp/app/src/main/java/com/egehanasal/advancedcontactsapp/ContactAdapter.java
package com.egehanasal.advancedcontactsapp;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.egehanasal.advancedcontactsapp.databinding.RecyclerRowBinding;

import java.util.ArrayList;

public class ContactAdapter extends RecyclerView.Adapter <ContactAdapter.ContactHolder> {

    private ArrayList<Contact> contactArrayList;

    public ContactAdapter(ArrayList<Contact> contactArrayList) {
        this.contactArrayList = contactArrayList;
    }

    @NonNull
    @Override
    public ContactHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        RecyclerRowBinding recyclerRowBinding = RecyclerRowBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new ContactHolder(recyclerRowBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ContactHolder holder, int position) {
        holder.binding.recyclerViewTextView.setText(contactArrayList.get(position).getName());
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(holder.itemView.getContext(), ContactActivity.class);
                intent.putExtra("info", "old");
                intent.putExtra("id", contactArrayList.get(position).id);
                holder.itemView.getContext().startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return contactArrayList.size();
    }

    public class ContactHolder extends RecyclerView.ViewHolder {
        private RecyclerRowBinding binding;

        public ContactHolder(RecyclerRowBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: kotlin/KotlinSQLite/app/src/main/java/com/egehanasal/kotlinsqlite/ContactAdapter.kt
package com.egehanasal.kotlinsqlite

import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.egehanasal.kotlinsqlite.databinding.RecyclerRowBinding

class ContactAdapter(val contactList: ArrayList<Contact>): RecyclerView.Adapter<ContactAdapter.ContactHolder>() {

    class ContactHolder(val binding: RecyclerRowBinding) : RecyclerView.ViewHolder(binding.root) {

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContactHolder {
        val binding = RecyclerRowBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ContactHolder(binding)
    }

    override fun onBindViewHolder(holder: ContactHolder, position: Int) {
        holder.binding.recyclerViewTextView.text = contactList[position].fullName
        holder.itemView.setOnClickListener {
            val intent = Intent(holder.itemView.context, DetailsActivity::class.java)
            intent.putExtra("info", "old")
            intent.putExtra("id", contactList.get(position).id)
            holder.itemView.context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return contactList.size
    }
}